### PR TITLE
Back out #62150 (Implement mem::{zeroed,uninitialized} in terms of MaybeUninit)

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
Ref: #62825

cc @RalfJung

### CODE REVIEW

File: src/libcore/intrinsics.rs [hunk 1]
```diff
@@ -702,16 +702,17 @@ extern "rust-intrinsic" {
...
     /// `Copy`, an all-zero value may not correspond to any legitimate
     /// state for the type in question.
-    #[unstable(feature = "core_intrinsics",
-               reason = "intrinsics are unlikely to ever be stabilized, instead \
-                         they should be used through stabilized interfaces \
-                         in the rest of the standard library",
-               issue = "0")]
-    #[rustc_deprecated(reason = "no longer used by rustc, will be removed - use MaybeUninit \
-                                 instead",
-                       since = "1.38.0")]
```
Comment (RalfJung) [inline src/libcore/intrinsics.rs]: We can keep it deprecated, should we not?

Comment (ishitatsuyuki) [reply src/libcore/intrinsics.rs]: It's an unstable item (that is intended for use inside rustc anyway), I don't have a strong opinion.

Comment (RalfJung) [reply src/libcore/intrinsics.rs]: Please keep the diff small by not removing the deprecation.

That should also make it easier alter to revert this PR.

Comment (RalfJung) [reply src/libcore/intrinsics.rs]: Actually it is probably best to instead *add* a deprecation mark at the other intrinsic. That most accurately reflects their status.

**Comment by ishitatsuyuki** [comment]
I went with an amend, so technically this is no longer a revert. Is that fine?

**Comment by ishitatsuyuki** [comment]
Out of time today, will fix it later.

File: src/libcore/intrinsics.rs [hunk 2]
```diff
@@ -702,15 +702,23 @@ extern "rust-intrinsic" {
     /// which is unsafe unless `T` is `Copy`. Also, even if T is
     /// `Copy`, an all-zero value may not correspond to any legitimate
     /// state for the type in question.
+    pub fn init<T>() -> T;
```
Comment (RalfJung) [inline src/libcore/intrinsics.rs]: This still de-deprecates `init`.

File: src/libcore/intrinsics.rs [hunk 3]
```diff
@@ -707,11 +707,26 @@ extern "rust-intrinsic" {
                          they should be used through stabilized interfaces \
                          in the rest of the standard library",
                issue = "0")]
-    #[rustc_deprecated(reason = "no longer used by rustc, will be removed - use MaybeUninit \
-                                 instead",
+    #[rustc_deprecated(reason = "superseded by MaybeUninit, removal planned",
```
Comment (RalfJung) [inline src/libcore/intrinsics.rs]: Why did you change the message? The new message is fine, but this will make reverting the commit more annoying.

Comment (ishitatsuyuki) [reply src/libcore/intrinsics.rs]: To be accurate. The comments and actual code should match, otherwise it would be confusing for someone who stumbled upon this.

Comment (RalfJung) [reply src/libcore/intrinsics.rs]: Fair.

**Comment by RalfJung** [comment]
Looking good, thanks a lot! Let's get this into the beta.

@bors r+ p=1
